kafka/source/pkg/reconciler/kafkasource.go

```diff
@@ -206,6 +219,188 @@ func checkResourcesStatus(src *v1alpha1.KafkaSource) error {
+	kedaPollingIntervalAnnodationKey     = kedaAutoscalingAnnotationClass + "/pollingInterval"
+	kedaTriggerLagThresholdAnnodationKey = kedaAutoscalingAnnotationClass + "/trigger.lagThreshold"
+)
+
+func convertMapKeyToInt32(dict map[string]string, key string, logger *zap.Logger) *int32 {
+	val, ok := dict[key]
+	if !ok {
+		return nil
+	}
+	i, err := strconv.ParseInt(val, 10, 32)
+	if err != nil {
+		logger.Error("Expected annotation value to be integer but got "+val, zap.Any("annotations key", key))
+		return nil
+	}
+	i32 := int32(i)
+	return &i32
+}
+
+func (r *Reconciler) readScalingAnnotations(ctx context.Context, src *v1alpha1.KafkaSource) {
+	logger := logging.FromContext(ctx).Desugar()
+	meta := src.GetObjectMeta()
+	annotations := meta.GetAnnotations()
+	if annotations != nil {
+		scalingClass := annotations[autoscaling.ClassAnnotationKey]
+		r.minReplicaCount = convertMapKeyToInt32(annotations, autoscaling.MinScaleAnnotationKey, logger)
+		r.maxReplicaCount = convertMapKeyToInt32(annotations, autoscaling.MaxScaleAnnotationKey, logger)
+		if scalingClass == kedaAutoscalingAnnotationClass {
+			r.cooldownPeriod = convertMapKeyToInt32(annotations, kedaCooldownPeriodAnnodationKey, logger)
+			r.pollingInterval = convertMapKeyToInt32(annotations, kedaPollingIntervalAnnodationKey, logger)
+			r.pollingInterval = convertMapKeyToInt32(annotations, kedaPollingIntervalAnnodationKey, logger)
+			r.triggerLagThreshold = convertMapKeyToInt32(annotations, kedaTriggerLagThresholdAnnodationKey, logger)
+
+		}
+	}
+}
+
+func (r *Reconciler) generateKedaScaledObjectUnstructured(ctx context.Context, ra *v1.Deployment, src *v1alpha1.KafkaSource) (*unstructured.Unstructured, error) {
+	logger := logging.FromContext(ctx).Desugar()
+	deploymentName := ra.GetName()
+	namespace := src.Namespace
+	name := src.Name
+	logger.Info("Unstructured ScaledObject name "+name, zap.Any("name", name))
+	srcName := src.GetName()
+	srcUID := src.GetUID()
+	srcResVersion := src.GetResourceVersion()
+	logger.Info("Got srcResVersion="+srcResVersion, zap.Any("srcResVersion", srcResVersion))
+	srcKind := src.GetGroupVersionKind().Kind
+	logger.Info("Got srcKind srcName srcUID", zap.Any("srcKind", srcKind), zap.Any("srcName", srcName), zap.Any("srcUID", srcUID))
+	srcBrokerList := src.Spec.BootstrapServers
+	srcConcumerGroup := src.Spec.ConsumerGroup
+	triggers := make([]map[string]interface{}, 0, 1)
+	topics := strings.Split(src.Spec.Topics, ",")
+	if len(topics) == 0 {
+		return nil, fmt.Errorf("Comma-separated list of topics can not be empty")
+	}
+	for _, topic := range topics {
+		triggerMetadata := map[string]interface{}{
+			"brokerList":    srcBrokerList,
+			"consumerGroup": srcConcumerGroup,
+			"topic":         topic,
+		}
+		if r.triggerLagThreshold != nil {
+			logger.Info("Got triggerLagThreshold", zap.Any("triggerLagThreshold", r.triggerLagThreshold))
+			triggerMetadata["lagThreshold"] = strconv.Itoa(int(*r.triggerLagThreshold))
```

Comment:
int will resolve to an integer of differently lengths on different arch systems, we've declared is specifically as a specific length earlier, shouldn't we use the same for the cast?

Reply:
fixed by using explicit format 64 call